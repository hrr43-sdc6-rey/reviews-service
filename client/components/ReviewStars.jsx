import React from 'react';

function ReviewStars({ stars }) {
  if (stars < 0.5) {
    return (
      <div>
        <p>&#11090;&#11090;&#11090;&#11090;&#11090;</p>
      </div>
    );
  }
  if (stars >= 0.5 && stars < 1.5) {
    return (
      <div>
        <p>&#11089;&#11090;&#11090;&#11090;&#11090;</p>
      </div>
    );
  }
  if (stars >= 1.5 && stars < 2.5) {
    return (
      <div>
        <p>&#11089;&#11089;&#11090;&#11090;&#11090;</p>
      </div>
    );
  }
  if (stars >= 2.5 && stars < 3.5) {
    return (
      <div>
        <p>&#11089;&#11089;&#11089;&#11090;&#11090;</p>
      </div>
    );
  }
  if (stars >= 3.5 && stars < 4.5) {
    return (
      <div>
        <p>&#11089;&#11089;&#11089;&#11089;&#11090;</p>
      </div>
    );
  }
  if (stars >= 4.5) {
    return (
      <div>
        <p>&#11089;&#11089;&#11089;&#11089;&#11089;</p>
      </div>
    );
  }
}

export default ReviewStars;
