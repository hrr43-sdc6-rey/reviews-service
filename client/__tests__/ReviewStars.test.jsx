import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import ReviewStars from '../components/ReviewStars.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('PageSelector', () => {
  it('PageSelector snapshot should match shallow render with no stars', () => {
    const stars = 0;
    const wrapper = shallow(
      <ReviewStars
        stars={stars}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector snapshot should match shallow render with 1 star', () => {
    const stars = 1;
    const wrapper = shallow(
      <ReviewStars
        stars={stars}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector snapshot should match shallow render with 2 stars', () => {
    const stars = 2;
    const wrapper = shallow(
      <ReviewStars
        stars={stars}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector snapshot should match shallow render with 3 stars', () => {
    const stars = 3;
    const wrapper = shallow(
      <ReviewStars
        stars={stars}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector snapshot should match shallow render with 4 stars', () => {
    const stars = 4;
    const wrapper = shallow(
      <ReviewStars
        stars={stars}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('PageSelector snapshot should match shallow render with 5 stars', () => {
    const stars = 5;
    const wrapper = shallow(
      <ReviewStars
        stars={stars}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
