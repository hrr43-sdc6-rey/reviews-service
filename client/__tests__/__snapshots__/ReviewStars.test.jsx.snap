// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSelector PageSelector snapshot should match shallow render with 1 star 1`] = `
<div>
  <p>
    ⭑⭒⭒⭒⭒
  </p>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with 2 stars 1`] = `
<div>
  <p>
    ⭑⭑⭒⭒⭒
  </p>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with 3 stars 1`] = `
<div>
  <p>
    ⭑⭑⭑⭒⭒
  </p>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with 4 stars 1`] = `
<div>
  <p>
    ⭑⭑⭑⭑⭒
  </p>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with 5 stars 1`] = `
<div>
  <p>
    ⭑⭑⭑⭑⭑
  </p>
</div>
`;

exports[`PageSelector PageSelector snapshot should match shallow render with no stars 1`] = `
<div>
  <p>
    ⭒⭒⭒⭒⭒
  </p>
</div>
`;
